BD-3329 Update FAQ about empty promotion codes (#7813)
braze-inc/braze-docs: _docs/_user_guide/personalization_and_dynamic_content/promotion_codes.md

### What happens when a promotion code list is expired or empty?

If the message should have contained a promotion code from an empty or expired list, the message will be canceled.

If the message contains Liquid logic that conditionally inserts a promotion code, the message will only be canceled if it should have contained a promotion code. If the message shouldn't have contained a promotion code, message will send normally.

> If the Liquid conditional is false and the message normally doesn't contain a promotion code, the message won't be canceled.

Author: ```suggestion
```

### What happens when a promotion code list is expired or empty?

> If the Liquid conditional in the message is true and results in sending a promotion code to the user, the message will be canceled. However, because promotion codes are "preloaded," the promotion code will still be used up, even though the message wasn't sent.

Author: ```suggestion
If the message should have contained a promotion code from an empty or expired list, the message will be canceled.

If the message contains Liquid logic that conditionally inserts a promotion code, the message will only be canceled if it should have contained a promotion code. If the message shouldn't have contained a promotion code, message will send normally.
```

> If the Liquid conditional is false and the email normally wouldn't contain a promotion code, the message won't be canceled.

Author: ```suggestion
If the Liquid conditional is false and the message normally wouldn't contain a promotion code, the message won't be canceled.
```

> If the Liquid conditional is false and the message normally doesn't contain a promotion code, the message won't be canceled.

Author: From @MarcusSchultheis:

Its a minor thing, but in the case that the promo code list is empty or expired, it doesn't make logical sense that "the promotion code will still be users up, even though the message wasn't sent". Perhaps something along the lines of:

If the message should have contained a promo code from a list that is empty or expired, the message will be aborted/canceled.
If the message contains some liquid logic that conditionally inserts a promo code, the message will only be aborted if it should have contained a promo code, if not, the message will send normally.

> If the Liquid conditional in the message is true and results in sending a promotion code to the user, the message will be canceled. However, because promotion codes are "preloaded", the promotion code will still be used up, even though the message wasn't sent.
> 
> If the Liquid conditional is false and the message normally wouldn't contain a promotion code, the message won't be canceled.

Reviewer: ```suggestion
If the Liquid conditional in the message is true and results in sending a promotion code to the user, the message will be canceled. However, because promotion codes are "preloaded," the promotion code will still be used up, even though the message wasn't sent.

If the Liquid conditional is false and the message normally doesn't contain a promotion code, the message won't be canceled.
```
